lib/algoliasearch-rails.rb

```diff
@@ -560,62 +507,80 @@ def algolia_reindex(batch_size = AlgoliaSearch::IndexSettings::DEFAULT_BATCH_SIZ
             group = group.select { |o| algolia_indexable?(o, tmp_options) }
           end
           objects = group.map { |o| tmp_settings.get_attributes(o).merge 'objectID' => algolia_object_id_of(o, tmp_options) }
-          tmp_index.save_objects(objects)
+
+          AlgoliaSearch.client.save_objects(tmp_index_name, objects)
         end
 
-        move_task = SafeIndex.move_index(tmp_index.name, src_index_name)
-        master_index.wait_task(move_task.raw_response["taskID"]) if synchronous || options[:synchronous]
+        task_id = AlgoliaSearch.client.operation_index(
+          tmp_index_name,
+          Algolia::Search::OperationIndexParams.new(operation: "move", destination: index_name)
+        ).task_id
+        AlgoliaSearch.client.wait_for_task(index_name, task_id) if synchronous || options[:synchronous]
       end
       nil
     end
 
     def algolia_set_settings(synchronous = false)
       algolia_configurations.each do |options, settings|
         if options[:primary_settings] && options[:inherit]
-          primary = options[:primary_settings].to_settings
+          primary = options[:primary_settings].to_settings.to_hash
           primary.delete :replicas
           primary.delete 'replicas'
-          final_settings = primary.merge(settings.to_settings)
+          final_settings = primary.merge(settings.to_settings.to_hash)
         else
-          final_settings = settings.to_settings
+          final_settings = settings.to_settings.to_hash
         end
 
-        index = SafeIndex.new(algolia_index_name(options), true)
-        task = index.set_settings(final_settings)
-        index.wait_task(task.raw_response["taskID"]) if synchronous
+        s = final_settings.map do |k, v|
+          [settings.setting_name(k), v]
+        end.to_h
+
+        synonyms = s.delete("synonyms") || s.delete(:synonyms)
+        unless synonyms.nil? || synonyms.empty?
+          resp = AlgoliaSearch.client.save_synonyms(index_name,synonyms.map {|s| Algolia::Search::SynonymHit.new({object_id: s.join("-"), synonyms: s, type: "synonym"}) } )
+          AlgoliaSearch.client.wait_for_task(index_name, resp.task_id) if synchronous
```

Comment:
should it be 
```suggestion
          AlgoliaSearch.client.wait_for_task(index_name, resp.task_id) if synchronous || options[:synchronous]
```
?

Reply:
It wasn't before, but it's probably good to add that here and on line 545 too. Adding it ✅